src/input/evdev.c

```diff
@@ -360,10 +417,48 @@ static bool evdev_handle_event(struct input_event *ev, struct input_device *dev)
         LiSendMouseButtonEvent(ev->value?BUTTON_ACTION_PRESS:BUTTON_ACTION_RELEASE, mouseCode);
         gamepadModified = false;
       } else if (gamepadCode != 0) {
-        if (ev->value)
+        if (ev->value) {
           dev->buttonFlags |= gamepadCode;
-        else
+          dev->btnDownTime = ev->time;
+        } else
           dev->buttonFlags &= ~gamepadCode;
+
+        if (gamepadCode == PLAY_FLAG && ev->value == 0) {
+          struct timeval elapsedTime;
+          timersub(&ev->time, &dev->btnDownTime, &elapsedTime);
+          int holdTimeMs = elapsedTime.tv_sec * 1000 + elapsedTime.tv_usec / 1000;
+          if (holdTimeMs >= MOUSE_EMULATION_LONG_PRESS_TIME) {
+            if (dev->meThread) {
+              pthread_cancel(dev->meThread);
+              dev->meThread = 0;
```

Comment:
You need to call `pthread_join(dev->meThread)` to avoid leaving zombie threads around.

`evdev_remove()` also needs to handle stopping the thread in case the device is removed while mouse emulation is enabled.